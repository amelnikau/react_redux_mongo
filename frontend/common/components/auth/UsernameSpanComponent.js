import React from 'react'
import {connect} from 'react-redux'

let UsernameSpanComponent = ({username}) => {
    return (
        <span>Username: {username}</span>
    )
};

const mapStateToProps = (state) => ({
    username: state.auth.username
});

UsernameSpanComponent = connect(mapStateToProps)(UsernameSpanComponent);

export default UsernameSpanComponent