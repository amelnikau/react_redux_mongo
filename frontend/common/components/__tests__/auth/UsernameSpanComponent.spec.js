import UsernameSpanComponent from '../../../components/auth/UsernameSpanComponent';
import renderer from 'react-test-renderer';
import React from 'react'
import configureMockStore from 'redux-mock-store'
import {Provider} from 'react-redux'

const mockStore = configureMockStore();

describe('Snapshot components testing', () => {
    describe('UsernameSpanComponent', () => {
        it("should render username test-user", () => {
            const tree = renderer.create(
                <Provider store={mockStore({auth: {username: 'test-user'}})}><UsernameSpanComponent/></Provider>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    })
});