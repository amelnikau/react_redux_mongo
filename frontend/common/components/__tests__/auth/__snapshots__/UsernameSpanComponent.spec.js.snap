// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot components testing UsernameSpanComponent should render username test-user 1`] = `
<span>
  Username: 
  test-user
</span>
`;
